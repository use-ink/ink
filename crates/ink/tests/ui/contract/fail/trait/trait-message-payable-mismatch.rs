#![allow(unexpected_cfgs)]

#[ink::trait_definition]
pub trait TraitDefinition {
    #[ink(message)]
    fn message(&self);
}

#[ink::contract]
mod contract {
    use super::TraitDefinition;

    #[ink(storage)]
    pub struct Contract {}

    impl Contract {
        #[ink(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }
    }

    impl TraitDefinition for Contract {
        #[ink(message, payable)]
        fn message(&self) {}
    }
}

fn main() {}
