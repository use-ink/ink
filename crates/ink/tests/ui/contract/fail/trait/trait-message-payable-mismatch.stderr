error[E0308]: mismatched types
  --> tests/ui/contract/fail/trait/trait-message-payable-mismatch.rs:25:9
   |
25 |         fn message(&mut self) {}
   |         ^^^^^^^^^^^^^^^^^^^^^^^^ expected `false`, found `true`
   |
   = note: expected struct `TraitMessagePayable<false>`
              found struct `TraitMessagePayable<true>`
